lib/vetext/service.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'vetext/configuration'
+
+module VEText
+  # Class used to connect to the VEText service which sends
+  # push notifications.
+  #
+  class Service < Common::Client::Base
+    configuration VEText::Configuration
+
+    BASE_PATH = '/api/vetext/pub/mobile/push'
+    REGISTER_PATH = BASE_PATH + '/endpoint'
+    PREFERENCES_PATH = BASE_PATH + '/preferences/client'
+    SEND_PATH = BASE_PATH + '/send'
+
+    # Register a user's mobile device with the push notification service.
+    #
+    # @app_name String       name from
+    # @device_token String   the unique token for the user's device
+    # @icn String            the Integration Control Number of the Veteran
+    # @os_info String        the operating system name from the device
+    # @device_name String    (optional) the name of the device
+    #
+    # @return Hash           response object, which includes endpoint_sid
+    #
+    def register(app_name, device_token, icn, os_info, device_name = nil)
+      app_sid = get_app_sid(app_name)
+      response = perform(
+        :put,
+        REGISTER_PATH, {
+          appSid: app_sid,
+          token: device_token,
+          icn: icn,
+          os: os_info[:name],
+          osVersion: os_info[:version],
+          deviceName: device_name || os_info[:name]
+        }
+      )
+      raise_if_response_error(response.body)
+      response.body
+    rescue Common::Client::Errors::ClientError => e
+      remap_error(e)
+    end
+
+    # Get preferences for a given user/device.
+    #
+    # @endpoint_sid String    the registration id as returned from `register`
+    #
+    # @return Hash            response object
+    #
+    def get_preferences(endpoint_sid)
+      response = perform(
+        :get,
+        "#{PREFERENCES_PATH}/#{endpoint_sid}",
+        nil
+      )
+      response.body
```

Comment:
`raise_if_response_error` needed here?

Reply:
The `success` key isn't part of the response for the `get_preferences` endpoint. I assumed that means there's no way to have a 200 response that's an error.

Reply:
True, good find.